rewrite-core/src/main/java/org/openrewrite/internal/StringUtils.java

```diff
@@ -369,4 +372,26 @@ public static String repeat(String s, int count) {
         System.arraycopy(multiple, 0, multiple, copied, limit - copied);
         return new String(multiple);
     }
+
+    private static boolean matchesGlob(@Nullable String value, @Nullable String globPattern) {
+        if ("*".equals(globPattern)) {
+            return true;
+        }
+        PathMatcher pm = FS.getPathMatcher("glob:" + globPattern);
+        Path path;
+        if (value != null && value.contains("/")) {

```

Comment:
On Windows paths may be delimited with "\", "/", or even a mixture of both. "C:/foo\bar/baz\" will be interpreted correctly and without complaint.

Now you might protest "But what madman would ever mix both??", and the answer is that Java's OS-specific utilities like those in java.nio will generally produce paths with "\" on Windows, but programmers will often hardcode "/" since that works everywhere. So in practice you end up with mixed-delimiter paths more often than you might assume.

Java utilities like `Path` are generally aware of this, so using a system-dependent `PathMatcher` is probably fine. But make sure the code you write around it understands this.

Reply:
In the use case I'm starting with, it's with XPath expression in YAML, so will only ever be `/`. If we run into a situation where we want to glob "real" paths, we can address that then IMO.